cpp/src/backends/core/backend.cc

```diff
@@ -1,6 +1,63 @@
 #include "src/backends/core/backend.hh"
 
+#include <memory>
+
+#include "src/backends/handler/handler_factory.hh"
+
 namespace torchserve {
+Backend::Backend() {}
+
+Backend::~Backend() {
+  handler_.reset();
+  model_instance_table_.clear();
+  // Todo: do proper cleanup
+  // dl_loader_->CloseDL();
+}
+
+bool Backend::Initialize(const std::string &model_dir) {
+  random_generator_.seed(time(0));
+  manifest_ = std::make_shared<torchserve::Manifest>();
+  // TODO: windows
+  if (!manifest_->Initialize(
+          fmt::format("{}/MAR-INF/MANIFEST.json", model_dir))) {
+    return false;
+  }
+
+  LoadHandler(model_dir);
+
+  if (!handler_) {
+    return false;
+  }
+
+  handler_->Initialize(model_dir, manifest_);
+
+  return true;
+}
+
+void Backend::LoadHandler(const std::string &model_dir) {
+  const std::string &handler_str = manifest_->GetModel().handler;
+  std::size_t delimiter_pos = handler_str.find(manifest_->kHandler_Delimiter);
+  if (delimiter_pos != std::string::npos) {
+#ifdef __APPLE__
```

Comment:
Will this require separate packaging for TorchServe Mac installables vs Linux version?

Reply:
We're currently not planning to provide precompiled binaries but will rely on the build.sh script for installation. If we change this in the future these macros will be resolved by the preprocessor during compilation and we would require different packages for the different platforms.

Reply:
We can handle this as a separate PR, filed issue  https://github.com/pytorch/serve/issues/2908 for tracking